services/laas/pipeline.go

```diff
@@ -0,0 +1,169 @@
+package laas
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	laas "github.com/ionos-cloud/sdk-go-laas"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+	"log"
+	"strings"
+)
+
+var pipelineResourceName = "LaaS Pipeline"
+
+func (c *Client) CreatePipeline(ctx context.Context, d *schema.ResourceData) (laas.Pipeline, utils.ApiResponseInfo, error) {
+	request := setPipelinePostRequest(d)
+	pipeline, apiResponse, err := c.sdkClient.PipelinesApi.PipelinesPost(ctx).Pipeline(*request).Execute()
+	apiResponse.LogInfo()
+	return pipeline, apiResponse, err
+}
+
+func (c *Client) IsPipelineAvailable(ctx context.Context, d *schema.ResourceData) (bool, error) {
+	pipelineId := d.Id()
+	pipeline, _, err := c.GetPipelineById(ctx, pipelineId)
+	if err != nil {
+		return false, err
+	}
+	if pipeline.Metadata == nil || pipeline.Metadata.Status == nil {
+		return false, fmt.Errorf("expected metadata, got empty for pipeline with ID: %s", pipelineId)
+	}
+	log.Printf("[DEBUG] pipeline status: %s", *pipeline.Metadata.Status)
+	return strings.EqualFold(*pipeline.Metadata.Status, "AVAILABLE"), nil
```

Comment:
utils.Available

Reply:
@cristiGuranIonos done in d8c4b47a210fc42a7af535eb2d82ab38246381bd